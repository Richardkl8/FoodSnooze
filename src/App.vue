<template>
    <div id="app">
        <custom-header></custom-header>
        <router-view/>
        <custom-footer></custom-footer>
    </div>
</template>

<script>
    import CustomHeader from '@/components/layout/CustomHeader.vue';
    import CustomFooter from '@/components/layout/CustomFooter.vue';

    export default {
        components: {
            CustomHeader,
            CustomFooter,
        },
    };
</script>

<style lang="scss">
    @import url('https://fonts.googleapis.com/css?family=Montserrat:400,700&display=swap');

    *,
    *::after,
    *::before {
        box-sizing: inherit;
        font-family: Montserrat, sans-serif;
    }

    html {
        height: 100%;
        width: 100%;
        font-size: 62.5%;

        @media only screen and (max-width: 900px) {
            font-size: 56.25%;
        }
        @media only screen and (max-width: 1200px) {
            font-size: 56.25%;
        }
        @media only screen and (min-width: 1800px) {
            font-size: 70%;
        }
    }

    body {
        height: 100%;
        width: 100%;
        font-family: Montserrat, serif;
        box-sizing: border-box;
    }

    #app {
        min-height: 100%;
        display: grid;
        grid-template-rows: auto 1fr auto;
        grid-template-columns: 100%;
    }

    .text-white {
        color: #fff;
    }

    .text-black {
        color: #071E22;
    }
</style>
