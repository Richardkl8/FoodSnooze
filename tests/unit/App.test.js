import { shallowMount } from '@vue/test-utils';
import App from '../../src/App.vue';

describe('App', () => {
    test('Should match snapShot', () => {
        const wrapper = shallowMount(App, {
            stubs: ['router-view'],
        });

        expect(wrapper).toMatchSnapshot();
    });

    test('Should show (stubbed) \'custom-header\', \'router-view\' \'custom-footer\' when mounted', () => {
        const wrapper = shallowMount(App, {
            stubs: ['router-view'],
        });

        expect(wrapper.html()).toContain(
            '<custom-header-stub></custom-header-stub>\n'
            + '  <router-view-stub></router-view-stub>\n'
            + '  <custom-footer-stub></custom-footer-stub>',
            );
    });
});
