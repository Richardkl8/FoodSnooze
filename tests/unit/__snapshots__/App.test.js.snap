// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapShot 1`] = `
<div id="app">
  <custom-header-stub></custom-header-stub>
  <router-view-stub></router-view-stub>
  <custom-footer-stub></custom-footer-stub>
</div>
`;
